tests/test_pygtracker.py

```diff
@@ -26,6 +28,71 @@ def test_version():
 
 # Start tests for rank_students
 
+# tests that the user inputs the right number of students
+def test_rank_students_num_of_students_match():
+    tracker = generate_input_calculate_final_grade()
+    assert (tracker.rank_students(n=4).shape[0]) == 4, "Number of students don't match"
+
+
+# raises an error when a course id is not a part of the dataset
+def test_rank_students_course_input_not_exist():
+    tracker = generate_input_calculate_final_grade()
+    with raises(NameError):
```

Comment:
As suggested in the main code
```suggestion
    with raises(ValueError):
```